include/arbor/simulation.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <array>
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include <arbor/common_types.hpp>
+#include <arbor/distributed_context.hpp>
+#include <arbor/domain_decomposition.hpp>
+#include <arbor/recipe.hpp>
+#include <arbor/sampling.hpp>
+#include <arbor/schedule.hpp>
+#include <arbor/util/handle_set.hpp>
+
+namespace arb {
+
+using spike_export_function = std::function<void(const std::vector<spike>&)>;
+
+struct simulation_state;
```

Comment:
A comment here along the lines of "forward declaration of pimpl implementation of simulation."

Reply:
Also a comment that mentions that we this is using PIMPL, and why.

Reply:
I'll add something brief in next tranche.